docs/proposal/content-library-support.md

```diff
@@ -0,0 +1,129 @@
+# Content Library Support
+
+```text
+---
+title: Support for Content Library
+authors:
+  - "@adam-jian-zhang"
+reviewers:  
+  - "@srm09"
+  - "@yastij"
+creation-date: 2023-03-27
+last-updated: 2023-03-27
+status: proposed
+---
+```
+
+## Table of Contents
+
+* [Content Library Support](#Content-Library-Support)
+  * [Table of Contents](#table-of-contents)
+  * [Glossary](#glossary)
+  * [Summary](#summary)
+  * [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+  * [Proposal](#proposal)
+    * [User Stories](#user-stories)
+      * [Story 1](#story-1)
+      * [Story 2](#story-2)
+    * [Overall Design](#overall-design)
+    * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+      * [Naming Scheme](#naming-scheme)
+      * [API changes](#api-changes)
+        * [VirtualMachineCloneSpec CR](#virtualmachineclonespec-cr)
+      * [Implementation Details](#implementation-details)
+        * [VSphereVM clone changes](#vspherevm-clone-changes)
+    * [Resources](#resources)
+
+## Glossary
+
+**Content Library**: The content library of vSphere, which stores artifacts such as OVA, vm template, etc. which can serve as the source the VM clones from.
+
+## Summary
+
+In current CAPV implementation, we only support cloning VM from vSphere template. This works fine for normal single vCenter
+scenario as everything is under a single vCenter context. If we consider multi-vCenter scenario, the template management becomes
+a problem, because we can not reference template cross vCenter boundary. We can manually copy the template to different vCenters as a workaround, but there is no mechanism to ensure the validity/consistency of the templates, it is error-prone and not maintainable.
+
+Content library can help template management in this case, it supports subscriptions among different content libraries and provides sync capability to help distribute OVA/templates to downstream content libraries. For example, we can set up a content library on the primary vCenter that hosts the management cluster, put the OVA/templates in the content library which serves as the version of truth for the OVA/templates we intended to use. And then for other vCenters that
+hosts workload clusters, we set up a content library and subscribe to the primary content library, and sync the content library items(templates/OVAs) to it. In this way it is easy to manage templates/OVAs and have confidence in the content of the templates/OVAs.
+
+Using content library as the source of templates has another little benefit that enables using local datastore as destination for the VMs, currently we are limited to shared datastore.
+
+## Motivation
+
+Enhance CAPV to clone VM from content library, to make multi-vCenter template management easier.
+
+### Goals
+
+* Provision VM from content library items(OVA or template).
```

Comment:
Maybe `Use ContentLibrary items as the VM image template`?

Also, should we call out somewhere if using ContentLibrary item turns the option to use linkedClone not possible anymore? (should be a restriction, I guess? or can we do linkedClone based on content library items?)